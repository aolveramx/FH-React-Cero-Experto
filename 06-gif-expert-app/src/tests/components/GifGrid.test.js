import '@testing-library/jest-dom'
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs')

describe('Tests in GifGrid.js', () => {

  const category = 'Dragon Ball'

  test('should render sucessful', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

    const wrapper = shallow(<GifGrid category={category} />)

    expect(wrapper).toMatchSnapshot()
    
  });

  test('should render items when image rendering useFetchGifs', () => {

    const gifs = [
    {
      id: 'ABC',
      url: 'http://localhost/cualquierCosa.jpg',
      title: 'cualquier cosa',
    },
    {
      id: 'CBA',
      url: 'http://localhost/cualquierCosa.jpg',
      title: 'cualquier cosa',
    }
  ]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })

    const wrapper = shallow(<GifGrid category={category} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

  });
  
});