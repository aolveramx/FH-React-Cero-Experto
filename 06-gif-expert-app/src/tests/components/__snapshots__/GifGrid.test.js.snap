// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifGrid.js should render items when image rendering useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="http://localhost/cualquierCosa.jpg"
    />
    <GifGridItem
      id="CBA"
      key="CBA"
      title="cualquier cosa"
      url="http://localhost/cualquierCosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Tests in GifGrid.js should render sucessful 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
